sdk/storage/faq.md

```diff
@@ -0,0 +1,374 @@
+   ```C++
+   try
+   {
+     // storage function goes here
+   }
+   catch (Azure::Storage::StorageException& e)
+   {
+     if (e.StatusCode == Azure::Core::Http::HttpStatusCode::ServiceUnavailable)
+     {
+       std::cout << e.Message << std::endl;
+     }
+   }
+   ```
+1. Store your data in multiple storage accounts for load balancing.
+1. Use CDN for static resources.
+
+Throttling error could also happen at subscription level or tenant level. You can still try above approaches or contact Azure Support in such cases.
+
+## How to troubleshoot 403 errors?
+
+403 error means your request to access Azure Storage is not correctly authorized.
+
+If you're using shared key authentication, you should
+
+1. Check the storage account name and account key match the resource you're trying to access, and the key is not rotated.
+1. If you need to add extra HTTP request headers or overwrite existing headers, do it with a HTTP pipeline policy, so that the new values can be signed with shared key.
+1. If you use a customized HTTP client, do not make any changes to HTTP request in the client.
+
+If you're using SAS authentication,
+
+1. Check the values of query parameter `st` and `se`, which specify a time range the SAS token is valid. Make sure the token is not expired.
+   If it's a user delegation SAS, also check `ske` which indicates the expiry of user delegation key.
+1. Make sure the shared key used to generate the SAS token is not rotated.
+1. If the SAS token is created with a stored access policy, make sure the access policy is still valid.
+1. SAS token has its own scope, for example it may be scoped to a storage account, a container or a blob/file. Make sure you don't access a resource out of the SAS token's scope.
+1. Check the message in exception.
+   You could print the information in exception with code below.
+   ```C++
+   try
+   {
+     // storage function goes here
+   }
+   catch (Azure::Storage::StorageException& e)
+   {
+     if (e.StatusCode == Azure::Core::Http::HttpStatusCode::Forbidden)
+     {
+       std::cout << e.RequestId << std::endl;
+       std::cout << e.ClientRequestId << std::endl;
+       std::cout << e.ErrorCode << std::endl;
+       std::cout << e.Message << std::endl;
+       for (const auto& i : e.AdditionalInformation)
+       {
+         std::cout << i.first << ":" << i.second << std::endl;
+       }
+     }
+   }
+   ```
+   For example, a common error code is `AuthorizationPermissionMismatch`, which means the SAS token doesn't have permission to perform this operation (like writing to a blob with a SAS token that only permits read).
+
+You shouldn't write your own function to generate signatures or sign the requests. It's not a trivial task and there are many edge cases to cover. Use Official tools or SDK instead.
+
+If you still cannot figure out the problem with above steps, you can open a GitHub issue with request ID or client request ID of the failed request.
```

Comment:
Is `client request ID` different from the `request ID?

```suggestion
If you still cannot figure out the problem with the above steps, you can open [a GitHub issue](https://github.com/Azure/azure-sdk-for-cpp/issues/new/choose) with the request ID or client request ID of the failed request.
```

Reply:
`x-ms-request-id` is generated and sent from server side. It can be used to uniquely identify a storage request.

`x-ms-client-request-id` is generated and sent from client side. It must be used together with storage account name, timestamp to identify a request.

`x-ms-client-request-id` is useful when a request is sent from client side but client doesn't receive a response (in this case we don't know `x-ms-request-id`).